### PR TITLE
Use Control.Monad.forever in main

### LINKED ISSUE #6: Use Control.Monad.forever in main
The watch mecanism is basically calling itself until the end of times, just like [Control.Monad.forever](https://hackage.haskell.org/package/base-4.10.1.0/docs/Control-Monad.html#v:forever).

Can you refactor it?

### PR DESCRIPTION
**Pull request by fdelsert**
resolve #6

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -4,18 +4,15 @@ import           TwitchAPI          (fetchViewerCount)
...
-
-watch :: Int -> IO () -> IO ()
-watch seconds action = do
-  action
-  putStrLn "."
-  threadDelay (toMicroseconds seconds)
-  watch seconds action
+  forever $ process (getOutputPath args) >> putStrLn "." >> wait 10
+
+wait :: Int -> IO()
```
Comment (ptitfred) [inline app/Main.hs]: :+1: it's a good idea to extract and express the idea

File: app/Main.hs [hunk 2]
```diff
@@ -4,18 +4,15 @@ import           TwitchAPI          (fetchViewerCount)
...
   args <- getArgs
-  watch 10 (process (getOutputPath args))
-
-watch :: Int -> IO () -> IO ()
-watch seconds action = do
-  action
-  putStrLn "."
-  threadDelay (toMicroseconds seconds)
-  watch seconds action
+  forever $ process (getOutputPath args) >> putStrLn "." >> wait 10
```
Comment (ptitfred) [inline app/Main.hs]: I would rather have kept the do block and wrap it with forever, something like this maybe:

```haskell
main :: IO ()
main = do
  args <- getArgs
  forever $ do
    process (getOutputPath args)
    putStrLn "."
    wait 10
```

File: app/Main.hs (hunk 1 above)
Comment (ptitfred) [reply app/Main.hs]: Keep the space between the type constructor `IO` and the type `()` :
```haskell
wait :: Int -> IO ()
```

**Review by ptitfred — CHANGES_REQUESTED** [review]
Thank you :)

Here you go with some remarks and suggestions.

File: app/Main.hs (hunk 2 above)
Comment (fdelsert) [reply app/Main.hs]: You are right, thanks for your advice.

**Review by ptitfred — APPROVED** [review]

**Pull request merged**